### ISSUE TITLE
Catalog queries should be retried

### ISSUE THREAD

**Issue by domodwyer** [influxdata/influxdb_iox#5008]
## TL;DR

All Catalog queries should retry when Postgres returns any error.

For example, Ingester can lose data due to transient errors; simply retry logic would fix this.

## Technical Description

Currently the ingester [does not](https://github.com/influxdata/influxdb_iox/blob/b8e7071f8229ec729369816430ac5c23d46b2f2e/ingester/src/data.rs#L143-L146) retry catalog queries (one example but there's many instances).

Once an error is returned from the internal buffer, the stream handler drops the write and continues consuming (incrementing the `potential_data_loss=true` metric):

https://github.com/influxdata/influxdb_iox/blob/b8e7071f8229ec729369816430ac5c23d46b2f2e/ingester/src/stream_handler/handler.rs#L343-L353

The buffer should retry non-fatal errors, and only return fatal ("this write will never succeed") errors to the stream handler.

There have been no incidence of errors causing data loss in the last 24h:

![image](https://user-images.githubusercontent.com/9275968/176700471-def2c914-91a4-4801-90c8-579262d80774.png)

Catalog errors are fortunately fairly rare, but induced when the catalog is under excessive load.

**Comment by crepererum** [comment]
We do that within the querier, so I would label this as https://github.com/influxdata/influxdb_iox/labels/Ingester as originally filed.

**Comment by jacobmarble** [comment]
@domodwyer please update status

**Comment by domodwyer** [comment]
Low risk and alerted on when it does happen - I intend to fix this by making the queries unnecessary via #4880. We could retry all query errors but we risk live-locking the ingester should a namespace/table/column not exist in the catalog.

#4880 is sufficiently close to being implemented, and this issue sufficiently rare, that it's not worth a dodgy intermediate fix.

**Comment by domodwyer** [comment]
This was resolved by #6069 🎉
